utils/money.ts

```diff
@@ -19,3 +19,43 @@ export const expressMillionsDollarsAsBillions = (
     unit: abbreviateUnits ? "B" : "billion",
   };
 };
+
+// Convert values expressed in millions of dollars to sensible human-readable precision with units
+// E.g. 1234567 -> { amount: "1.2", unit: "trillion" }
+export const abbreviateMillionsDollars = (
+  millionsDollars: number,
+  abbreviateUnits: boolean = false,
+  precision?: number,
+): {
+  amount: string
+  unit: string
+} => {
+  if (Math.abs(millionsDollars) < 1) {
+    return {
+      amount: "<$1",
+      unit: abbreviateUnits ? "M" : "million",
+    };
+  }
+  const dollars = millionsDollars * 1_000_000;
+  const [amount, unitAbbr] = new Intl.NumberFormat(undefined, {
+    notation: "compact",
+    maximumFractionDigits: precision ?? 1,
+    minimumFractionDigits: precision ?? 1,
+    style: "currency",
+    currency: "USD",
+    currencyDisplay: "narrowSymbol",
+  }).format(dollars).split(/([KMBT])/);
+  let unit;
+  if (!abbreviateUnits) {
+    unit = {
+      K: "thousand",
```

Comment:
I might be reading this wrong, but is it not the case that you'll never hit the K case because earlier you returned a flat <$1M for values less than 1 million?